drivers/hmis/app/models/hmis/hud/enrollment.rb

```diff
@@ -55,6 +55,13 @@ class Hmis::Hud::Enrollment < Hmis::Hud::Base
     left_outer_joins(:wip).where(editable_wip.or(editable_enrollment))
   end
 
+  scope :in_project_including_wip, ->(project) do
+    wip_enrollments = wip_t[:project_id].in(project.id)
+    actual_enrollments = e_t[:ProjectID].in(project.project_id)
+
+    left_outer_joins(:wip).where(wip_enrollments.or(actual_enrollments))
+  end
```

Comment:
I don't love that we are passing the whole project record, but if we were to pass just the id it would be an extra lookup.

To get around the need for 2 ids, we could join the enrollments table with the projects table so that we can look up purely by `id`.

Or, we could revisit the decision to use `id` instead of `project_id` in the WIP table. Maybe it wasn't the right call.

Reply:
I actually don't mind passing the whole project object.  Using the object instead of the values means we can call `preload(:project)` or similar to avoid extra queries.  I can't remember the details about why we went with `project_id` If I remember correctly, it allows us to create stub project records with no ProjectID so they don't interfere with the warehouse data until they're no longer WIP.  I don't think there's going to be a way to store the `id` on the enrollment side unless we guarantee `project_id` and `id` are the same.

Reply:
Looking at https://github.com/greenriver/hmis-warehouse/pull/2756/files#r1070180815, maybe it would make sense to do:
```suggestion
  scope :in_project_including_wip, ->(project_ids, wip_project_ids) do
    wip_enrollments = wip_t[:project_id].in(Array.wrap(wip_project_ids))
    actual_enrollments = e_t[:ProjectID].in(Array.wrap(project_ids))

    left_outer_joins(:wip).where(wip_enrollments.or(actual_enrollments))
  end
```
That way you could do
```
  def enrollments
     Hmis::Hud::Enrollment.in_project_including_wip(id, project_id)
   end
```
over on the project side, and potentially, if you wanted to know about enrollments across projects, you could pass arrays of the various id types.  That said, not sure that's any better really.